### PR TITLE
luci-admin-full: add wifi option

### PR DESCRIPTION
**Pull request by Ansuel**
This commit adds option to disable scan for 40mhz channel, permit to tweak beacon interval and other advanced configuration.

Based on this #1699

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

### CODE REVIEW

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua [hunk 1]
```diff
@@ -250,6 +250,13 @@ if hwtype == "mac80211" then
 
 	s:taboption("advanced", Value, "frag", translate("Fragmentation Threshold"))
 	s:taboption("advanced", Value, "rts", translate("RTS/CTS Threshold"))
+	
+	s:taboption("advanced", Flag, "noscan", translate("Force 40MHz mode"),"Always use 40MHz channels even if the secondary channel overlaps.\
+				Using this option does not comply with IEEE 802.11n-2009!").optional = true
+				
+	beacon_int = s:taboption("advanced", Value, "beacon_int", translate("Beacon Interval"),"(15 - 65535)")
+        beacon_int.optional = true
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua]: please indent using tabs

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua [hunk 2]
```diff
@@ -250,6 +250,13 @@ if hwtype == "mac80211" then
 
 	s:taboption("advanced", Value, "frag", translate("Fragmentation Threshold"))
 	s:taboption("advanced", Value, "rts", translate("RTS/CTS Threshold"))
+	
+	s:taboption("advanced", Flag, "noscan", translate("Force 40MHz mode"),"Always use 40MHz channels even if the secondary channel overlaps.\
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua]: Wrap the `Always use...` text into `translate()`

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua [hunk 3]
```diff
@@ -491,6 +498,17 @@ if hwtype == "mac80211" then
 
 	ifname = s:taboption("advanced", Value, "ifname", translate("Interface name"), translate("Override default interface name"))
 	ifname.optional = true
+	
+	short_preamble = s:taboption("advanced", Flag, "short_preamble", translate("Short Preamble"))
+    short_preamble.default = short_preamble.enabled
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua]: please indent using tabs

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua [hunk 4]
```diff
@@ -491,6 +498,17 @@ if hwtype == "mac80211" then
...
+
+    dtim_period = s:taboption("advanced", Value, "dtim_period", translate("DTIM Interval"),"Delivery Traffic Indication Message Interval")
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua]: Please wrap the `Deliver Traffic ...` text into `translate()` as well and add `dtim_period` to the list of local variables at the top

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua [hunk 5]
```diff
@@ -491,6 +498,17 @@ if hwtype == "mac80211" then
...
+    dtim_period.optional = true
+    dtim_period.placeholder = 2
+
+    disassoc_low_ack = s:taboption("advanced", Flag, "disassoc_low_ack", translate("Disassociate On Low Acknowledgement"),
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua]: Please indent using tabs and add `disassoc_low_ack` to the list of local variables at the top

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua [hunk 6]
```diff
@@ -491,6 +498,17 @@ if hwtype == "mac80211" then
...
+															"Allow AP mode to disconnect STAs based on low ACK condition")
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua]: Please remove this empty line

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua [hunk 7]
```diff
@@ -250,6 +250,13 @@ if hwtype == "mac80211" then
 
 	s:taboption("advanced", Value, "frag", translate("Fragmentation Threshold"))
 	s:taboption("advanced", Value, "rts", translate("RTS/CTS Threshold"))
+	
+	s:taboption("advanced", Flag, "noscan", translate("Force 40MHz mode"),"Always use 40MHz channels even if the secondary channel overlaps.\
+				Using this option does not comply with IEEE 802.11n-2009!").optional = true
+				
+	beacon_int = s:taboption("advanced", Value, "beacon_int", translate("Beacon Interval"),"(15 - 65535)")
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua]: Add `beacon_int` to the list of local variables at the top

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua [hunk 8]
```diff
@@ -491,6 +498,17 @@ if hwtype == "mac80211" then
 
 	ifname = s:taboption("advanced", Value, "ifname", translate("Interface name"), translate("Override default interface name"))
 	ifname.optional = true
+	
+	short_preamble = s:taboption("advanced", Flag, "short_preamble", translate("Short Preamble"))
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua]: Add `short_preamble` to the list of variables at the top

**Comment by Ansuel** [comment]
ok should be good now @jow-

**Comment by jow-** [comment]
Merged with further fixes in 39cabc036

**Pull request closed without merging**